tools/bio3d/dccm.xml

```diff
@@ -0,0 +1,131 @@
+<tool id="bio3d_dccm" name="DCCM analysis" version="@VERSION@">
+    <description>Dynamical Cross-Correlation Maps using Bio3D (DCCM)</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code">
+<![CDATA[ 
+   Rscript '$__tool_directory__/dccm.R'
+        '$dcdin' 
+        '$pdbin'
+        #if $dccm.sele == 'calpha':
+          "string"
+           "calpha"
+        #end if
+        #if $dccm.sele == 'cbeta':
+          "string"
+          '$dccm.cbeta'
+        #end if
+        #if $dccm.sele == 'backbone':
+          "string"
+          "backbone"
+        #end if
+        #if $dccm.sele == 'sidechain':
+          "string"
+          "sidechain"
+        #end if
+        #if $dccm.sele == 'protein':
+          "string"
+          "protein"
+        #end if
+        #if $dccm.sele == 'ligand':
+          "string"
+          "ligand"
+        #end if
+        #if $dccm.sele == 'nucleic':
+          "string"
+          "nucleic"
+        #end if
+        #if $dccm.sele == 'elety':
+          "elety"
+          '$dccm.elety'
+        #end if
+        #if $dccm.sele == 'resid':
+          "resid"
+          '$dccm.resid'
+        #end if
+        #if $dccm.sele == 'segid':
+          "segid"
+          '$dccm.segid'
+        #end if
+        '$output'
+        '$dccm_plot'
+    2>&1
+]]></command>
+    <inputs>
+        <expand macro="analysis_inputs"/>
+        <conditional name="dccm">
+           <param name="sele" type="select" label="Select domains">
+             <option value="calpha">Calpha</option>
+             <option value="cbeta">Cbeta</option>
+             <option value="backbone">Backbone</option>
+             <option value="sidechain">Sidechain</option>
+             <option value="protein">Protein</option>
+             <option value="ligand">Ligand</option>
+             <option value="nucleic">Nucleic Acids</option>
+             <option value="elety">Atom Names</option>
+             <option value="resid">Resid</option>
+             <option value="segid">Segid</option>
+           </param>
+           <when value="calpha"/>
+           <when value="cbeta"/>
+           <when value="backbone"/>
+           <when value="sidechain"/>
+           <when value="protein"/>
+           <when value="ligand"/>
+           <when value="nucleic"/>
+           <when value="elety">
+                <param name="elety"  type="text" value="CA" label="Atom Name"/>
+            </when>
+            <when value="resid">
+                <param name="resid"  type="text" value="BGLC" label="Resid"/>
+            </when>
+            <when value="segid">
+                <param name="segid"  type="text" value="SUBS" label="Segid"/>
```

Comment:
Can you please add validators to all the text inputs and only allow letters?

Reply:
@bgruening this is a way to select different parts of the system (Different chains, different ligands). These are normally defined letters. You can also use numbers but normally in a pdb you only use letters.

Reply:
We can also use a validator for letters-and-numbers.